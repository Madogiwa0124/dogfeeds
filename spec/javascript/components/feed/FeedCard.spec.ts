import { mount } from "@vue/test-utils";
import Component from "@js/components/feed/FeedCard.vue";
import { buildFeed } from "@spec/mocks/Feed";
import { advanceTo } from "jest-date-mock";

describe("components/feed/FeedCard.vue", () => {
  it("snapshot", () => {
    const props = { feed: buildFeed(1) };
    advanceTo(new Date("2021-05-03T15:35:47+09:00"));
    const wrapper = mount(Component, { propsData: props });
    expect(wrapper.element).toMatchSnapshot();
  });
});
