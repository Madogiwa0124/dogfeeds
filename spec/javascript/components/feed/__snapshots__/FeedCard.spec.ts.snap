// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/feed/FeedCard.vue snapshot 1`] = `
<div
  class="feed-card card"
  id="feed-card-1"
>
  <a
    href="/feeds/1"
    rel=""
    target="_self"
  >
    <header
      class="card-header"
    >
      <p
        class="card-header-title"
      >
        feed title 1
      </p>
    </header>
    <div
      class="card-content"
    >
      <div
        class="card-image"
      >
        <img
          alt="entry title 1"
          height="220"
          src="https://example/com/sample.png"
          width="350"
        />
      </div>
      <div
        class="content"
      >
        <div
          class="tag-area field is-grouped is-grouped-multiline"
        >
          <!--v-if-->
          
          <div
            class="control"
          >
            <div
              class="my-tag tags has-addons"
            >
              <span
                class="tag is-success"
              >
                feed tag body 1
              </span>
              <!--v-if-->
            </div>
          </div>
          <div
            class="control"
          >
            <div
              class="my-tag tags has-addons"
            >
              <span
                class="tag is-success"
              >
                feed tag body 1
              </span>
              <!--v-if-->
            </div>
          </div>
          
        </div>
        <object>
          <a
            class="has-text-info last-entry-title"
            href="https://example.com"
            rel="noopener"
            target="_blank"
          >
            entry title 1 
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-external-link-alt fa-w-16"
              data-icon="external-link-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                fill="currentColor"
              />
            </svg>
          </a>
        </object>
        <p
          class="entry-info has-text-right"
        >
          <span
            title="2021/05/03 2:35:46"
          >
            Last updated an hour ago
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-clock fa-w-16"
            data-icon="clock"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class=""
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
              fill="currentColor"
            />
          </svg>
          <span
            class="entry-clip clip"
            title="Clipして後で読む"
          >
            
            
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-paperclip fa-w-14"
              data-icon="paperclip"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M43.246 466.142c-58.43-60.289-57.341-157.511 1.386-217.581L254.392 34c44.316-45.332 116.351-45.336 160.671 0 43.89 44.894 43.943 117.329 0 162.276L232.214 383.128c-29.855 30.537-78.633 30.111-107.982-.998-28.275-29.97-27.368-77.473 1.452-106.953l143.743-146.835c6.182-6.314 16.312-6.422 22.626-.241l22.861 22.379c6.315 6.182 6.422 16.312.241 22.626L171.427 319.927c-4.932 5.045-5.236 13.428-.648 18.292 4.372 4.634 11.245 4.711 15.688.165l182.849-186.851c19.613-20.062 19.613-52.725-.011-72.798-19.189-19.627-49.957-19.637-69.154 0L90.39 293.295c-34.763 35.56-35.299 93.12-1.191 128.313 34.01 35.093 88.985 35.137 123.058.286l172.06-175.999c6.177-6.319 16.307-6.433 22.626-.256l22.877 22.364c6.319 6.177 6.434 16.307.256 22.626l-172.06 175.998c-59.576 60.938-155.943 60.216-214.77-.485z"
                fill="currentColor"
              />
            </svg>
          </span>
        </p>
      </div>
    </div>
    <footer
      class="card-footer"
    >
      <p
        class="card-footer-item"
      >
        <button
          class="select-feed button is-conversion is-rounded is-outline selected-feed"
        >
          <span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-square fa-w-14"
              data-icon="square"
              data-prefix="far"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
                fill="currentColor"
              />
            </svg>
             RSSフィードを選択する 
          </span>
        </button>
      </p>
    </footer>
  </a>
</div>
`;
